### ISSUE TITLE
Hard dependency on "sweetalert2" package

### ISSUE THREAD

**Issue by zenflow** [zenflow/sweetalert2-react#1]
https://github.com/zenflow/sweetalert2-react/blob/c23ec32540cf13d0d650cd75327f070bf71efefc/src/SweetAlert.js#L3
https://github.com/zenflow/sweetalert2-react/blob/c23ec32540cf13d0d650cd75327f070bf71efefc/package.json#L64-L70

More explanation later if needed

**Comment by toverux** [comment]
It can be a peer dep with `sweetalert2` marked as an "external" dependency in Webpack, so it's not included in the library build. Is that the problem here?

**Comment by zenflow** [comment]
No, there is no library build here. But thanks for the tip! I needed that knowledge for sweetalert2-react-content!

Issue is that `swal` is always `require('sweetalert2')`, and in order to use this with sweetalert2-react-content we need to be able to redefine `swal` as `require('sweetalert2-react-content')(require('sweetalert2'))`

Should be an easy change to support optionally passing in your own `swal` instance.

**Comment by zenflow** [comment]
@toverux Sorry, yes, `sweetalert2` being a regular dependency instead of a peer dependency is part of the problem.
